keps/sig-cli/5295-kyaml/README.md

```diff
@@ -0,0 +1,1500 @@
+YAML calls this “flow style”,
+as opposed to the conventional “block style”.
+
+This might sound a lot like JSON,
+because it is!
+Remember that JSON is valid YAML.
+KYAML is friendlier than JSON for humans
+to read,
+to write,
+and to commit to source control.
+Unlike JSON, KYAML:
+1. Allows comments (when authored by users).
+2. Allows trailing commas on the last item in a list or struct/map.
+3. Does not require quoted keys.
+
+KYAML also includes a “header” (still valid YAML)
+which helps to disambiguate a KYAML document from an ill-formed JSON document
+(see: [kubernetes/kubernetes#77946](https://github.com/kubernetes/kubernetes/issues/78946)).
+
+KYAML also tries to economize on vertical space by ["cuddling"](https://blog.gskinner.com/archives/2008/11/curly_braces_to.html) some kinds of brackets together.
+
+Because KYAML is YAML, a KYAML multi-document is a YAML multi-document.
+
+To make this change, we will:
+1. Create new code (see details below) which serializes any object into KYAML.
+2. Introduce a kubectl output format, `-o=kyaml`, which uses this new package.
+3. Create tooling which reformats YAML or JSON as KYAML (see details below).
+4. Work with SIG Docs to update examples to this format, both example command-lines and examples of API objects.
+5. Look for other examples across kubernetes-affiliated git repositories, and offer PRs to convert to KYAML.
+6. Make a lot of noise about it, and try to get more people to use it
+
+The rationale for changing everything everywhere is simple.
+Many users start with examples,
+which they copy and modify.
+By changing the examples we seed the change in the ecosystem.
+One success metric might be
+that KYAML becomes the preferred format in the most common helm charts.
+
+### Notes/Constraints/Caveats
+
+**Important note:**
+This is EXCLUSIVELY about client-side formatting.
+This KEP *does not* propose any changes to apiserver
+or other components besides kubectl.
+
+Syntax has always been a magnet for arguments in our industry.
+We expect this to be no exception.
+Reasonable people will disagree with our choices,
+which is fine, but we should hold a strong opinion on this.
+
+YAML is a very large and complex specification.
+We expect that we will find places where KYAML may be less readable than conventional YAML.
+We should revisit this specification when that happens.
+Given that KYAML is YAML it should be possible to revise
+without breaking anyone.
+
+### Risks and Mitigations
+
+Risk: Our encoder produces some output which is not valid YAML.
+Mitigation: That’s a bug. It’s hard to imagine a case that can’t be fixed.
+
+Risk: The result could be net worse than plain YAML.
+Mitigation: We will produce many examples and include them in the review process. We will solicit feedback from various groups on the result.
+
+Risk: This gives a false sense of safety, but doesn’t SOLVE  the problems with YAML, just avoids them.
+Mitigation: Docs?
```

Comment:
Maybe the right answer here should be that our goal is not to modify YAML format, but rather focus on a valid YAML sub-set, which resolves known problems, increasing the success of our users.

Reply:
I took your words, thanks.